### PR TITLE
Loops

### PR DESCRIPTION
**Pull request by drozdi-k**

### CODE REVIEW

File: src/main/java/Loops/OddNumbers/DoWhileOddChecker.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package Loops.OddNumbers;
```
Comment (drozdi-k) [inline src/main/java/Loops/OddNumbers/DoWhileOddChecker.java:1]: Разработать классы ForOddChecker, WhileOddChecker и DoWhileOddChecker, каждый из котороых выводит на экран нечетные числа в диапазоне от 0 до 50 включительно используя цикл, который соответствует названию класса (for, while, и do .. while).

Решение можно реализовать в методе main() соответствующего класса.

File: src/main/java/Loops/PrimeNumbers.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
+package Loops;
```
Comment (drozdi-k) [inline src/main/java/Loops/PrimeNumbers.java:1]: Разработать класс PrimeNumbers, который считает сумму простых чисел в диапазоне от 1 до 500 включительно. Дополнительно посчитайте количество простых чисел, которые входят в этот диапазон.

Ограничения:

Числа 17 и 71 не должны учитываться при подсчете суммы и количества простых чисел.
В случае, если количество простых чисел равно 50, подсчет необходимо завершить досрочно.
Для решения задачи используйте цикл for и операторы continue, break.
Необходимо вывести на экран количество простых чисел и их сумму.

Решение можно реализовать в методе main() соответствующего класса.

File: src/main/java/Loops/Car.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
+package Loops;
```
Comment (drozdi-k) [inline src/main/java/Loops/Car.java:1]: Разработать класс представляющий автомобиль, соблюдая инкапсуляцию, который обладает следующими характеристиками:

Свойства

Модель
Цвет
Максимальная скорость
Текущая скорость
Методы

Ускориться: void accelerate(int targetSpeed) - должен увеличивать скорость автомобиля используя цикл. В каждой итерации скорость нужно увеличивать на 1 ед. до достижения заданной скорости или максимально допустимой скорости автомобиля. Если текущая скорость автомобиля больше заданной, то автомобиль ускоряться не должен.
Замедлиться: void decelerate(int targetSpeed) - должен уменьшать скорость автомобиля используя цикл. В каждой итерации скорость нужно уменьшать на 1 ед. до достижения заданной скорости или полной остановки автомобиля. Если текущая скорость меньше заданной, то автомобиль замедляться не должен.
Находится ли автомобиль в движении: boolean isDriving() - возвращает true если скорость автомобиля больше 0.
Стоит ли автомобиль на месте: boolean isStopped() - возвращает true если скорость автомобиля равна 0.
Есть ли возможность ускориться: boolean canAccelerate() - возвращает true если текущая скорость меньше максимальной.
Модель, цвет и максимальная скорость должны задаваться через конструктор. Текущая скорость может быть изменена только путем вызова основных методов.

Добавить вспомогательные getter методы, переопределить методы equals() и toString().

File: src/main/java/Arrays/ArraysService.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    public static void printArray(int[] array) {  //распечатать логгерами походу
+        for (int i = 0; i < array.length; i++) {
+             logger.info("array[" + i + "] = " + array[i]);
+        }
+    }
+
+    public static int sum(int[] array) {
+        int sum = 0;
+        for (int j : array) {
+            sum = sum + j;
```
Comment (Denis-Pikus) [inline src/main/java/Arrays/ArraysService.java:31]: better use the notation "sum += j"

File: src/main/java/Arrays/Palindrome.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+        for(int i=0,j=len-1; i < len/2;i++,j--){
+            char temp = arrayReversed[i];
+            arrayReversed[i] = arrayReversed[j];
+            arrayReversed[j] = temp;
+        }
+        String s1 = String.valueOf(array);
+        String s2 = String.valueOf(arrayReversed);
+        logger.info("original: " + s1);
+        logger.info("reversed: " + s2);
```
Comment (Denis-Pikus) [inline src/main/java/Arrays/Palindrome.java:25]: You can put an array into a logger without converting to a string

File: src/main/java/Collections/Book.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Book book = (Book) o;
+        return Objects.equals(author, book.author) && Objects.equals(name, book.name);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(author, name);
+    }
```
Comment (Denis-Pikus) [inline src/main/java/Collections/Book.java:41]: Observe the structure of the class. Constructors come first, then getters and setters, then hashCode and equals, then toString.

File: src/main/java/Collections/Library.java [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+        for (Book book: library) {
+
+            if (book.getAuthor().toLowerCase().equals(author.toLowerCase())
+                    && !author.equals("")) {
+                find.add(book);
+            }
+        }
+        logger.info(find);
+        return find;
+    }
```
Comment (Denis-Pikus) [inline src/main/java/Collections/Library.java:62]: Observe the structure of the class. Constructors come first, then getters and setters, then hashCode and equals, then toString.

File: src/test/java/MathTest.java [hunk 8]
```diff
@@ -1,35 +1,73 @@
...
+    }
+
+    @Test
+    public void antiMultiplication() {
+        int a = 22;
+        int b = 2;
+        logger.info("a * b = " + a + " * " + b + " = " + Math.multiplication(a,b));
+        assertEquals(27, Math.multiplication(a,b));
     }
 }
```
Comment (Denis-Pikus) [inline src/test/java/MathTest.java:73]: You need to add a test to check division by zero

**Review by Denis-Pikus — APPROVED** [review]
lgfm

**Review by integratorBY — APPROVED** [review]
**lgfm**

**Review by integratorBY — APPROVED** [review]
**lgfm**

**Review by integratorBY — APPROVED** [review]
**lgfm**

**Review by integratorBY — APPROVED** [review]
**lgfm**

**Review by olgaZskiba — APPROVED** [review]
lgfm
